src/security-council-mgmt/SecurityCouncilManager.sol

```diff
@@ -207,14 +220,55 @@ contract SecurityCouncilManager is
     }
 
     /// @inheritdoc ISecurityCouncilManager
-    function rotateMember(address _currentAddress, address _newAddress)
-        external
-        onlyRole(MEMBER_ROTATOR_ROLE)
-    {
-        Cohort cohort = _swapMembers(_currentAddress, _newAddress);
+    function recoverAddMemberMessage(uint256 nonce, bytes calldata signature) public view returns (address) {
+        bytes32 digest = _hashTypedDataV4(keccak256(abi.encode(
+            keccak256("addMember(uint256 nonce)"),
+            nonce
+        )));
+        return ECDSAUpgradeable.recover(digest, signature);
+    }
```

Comment:
We should include more information in the signature. If the member uses the wrong nonce to sign then this will return a different address that has an unknown key.

```suggestion
    function recoverRotateMemberMessage(uint256 nonce, address oldMember, address newMember, bytes calldata signature) public view returns (address) {
        bytes32 digest = _hashTypedDataV4(keccak256(abi.encode(
            keccak256("RotateMember(uint256 nonce,address oldMember,address newMember)"),
            nonce,
            oldMember,
            newMember
        )));
        return ECDSAUpgradeable.recover(digest, signature);
    }
```

and then in `rotateMember` we should have something like this, where newMember is a function argument.

```
address newAddress = recoverRotateMemberMessage(updateNonce, msg.sender, newMember, signature);
require(newAddress == newMember, "Invalid signature");
```

Reply:
Added those.

I didnt add newMember to the sig, since it's being recovered and checked.